PJ05/src/Exceptions/InvalidRequestException.java

```diff
@@ -2,16 +2,13 @@
 
 import Request.Request;
 
-public class InvalidRequestException extends Exception
-{
+public class InvalidRequestException extends RequestFailedException {
```

Comment:
This exception could be regarded as RequestFailedException. It could also be regarded as a system error, since the Client fails to send a proper request. The decision is left to you.